code/parachain/runtime/common/src/fees.rs

```diff
@@ -0,0 +1,162 @@
+use crate::{prelude::*, Balance};
+use composable_support::math::safe::safe_multiply_by_rational;
+use composable_traits::{
+	currency::{AssetExistentialDepositInspect, AssetRatioInspect, Rational64},
+	rational,
+};
+
+use frame_support::{
+	traits::ConstU128,
+	weights::{
+		constants::ExtrinsicBaseWeight, WeightToFeeCoefficient, WeightToFeeCoefficients,
+		WeightToFeePolynomial,
+	},
+};
+use num_traits::One;
+use primitives::currency::CurrencyId;
+use sp_runtime::Perbill;
+use sp_std::marker::PhantomData;
+
+pub const NATIVE_EXISTENTIAL_DEPOSIT: NativeBalance = 100_000_000_000;
+pub type NativeExistentialDeposit = ConstU128<NATIVE_EXISTENTIAL_DEPOSIT>;
+
+pub struct WeightToFeeConverter;
+impl WeightToFeePolynomial for WeightToFeeConverter {
+	type Balance = Balance;
+	fn polynomial() -> WeightToFeeCoefficients<Self::Balance> {
+		let p = CurrencyId::milli::<Balance>();
+		let q = 10 * Balance::from(ExtrinsicBaseWeight::get());
+		smallvec::smallvec![WeightToFeeCoefficient {
+			degree: 1,
+			negative: false,
+			coeff_frac: Perbill::from_rational(p % q, q),
+			coeff_integer: p / q,
+		}]
+	}
+}
+
+#[cfg(feature = "runtime-benchmarks")]
+pub fn multi_existential_deposits<AssetsRegistry>(_currency_id: &CurrencyId) -> Balance {
+	// ISSUE:
+	// Running benchmarks with non zero multideposit leads to fail in 3rd party pallet.
+	// It is not clearly why it happens.pub const BaseXcmWeight: Weight = 100_000_000;
+	// 2022-03-14 20:50:19 Running Benchmark: collective.set_members 2/1 1/1
+	// Error:
+	//   0: Invalid input: Account cannot exist with the funds that would be given
+	use num_traits::Zero;
+	Balance::zero()
+}
+
+#[cfg(not(feature = "runtime-benchmarks"))]
+pub fn multi_existential_deposits<
+	AssetsRegistry: AssetRatioInspect<AssetId = CurrencyId>
+		+ AssetExistentialDepositInspect<AssetId = CurrencyId, Balance = Balance>,
+>(
+	currency_id: &CurrencyId,
+) -> Balance {
+	AssetsRegistry::existential_deposit(*currency_id)
+		.ok()
+		.or(WellKnownPriceConverter::existential_deposit(*currency_id))
+		.unwrap_or(Balance::MAX)
+}
+
+pub struct PriceConverter<AssetsRegistry>(PhantomData<AssetsRegistry>);
+
+pub mod cross_chain_errors {
+	pub const ASSET_IS_NOT_PRICEABLE: &str = "Asset is not priceable";
+	pub const AMOUNT_OF_ASSET_IS_MORE_THAN_MAX_POSSIBLE: &str =
+		"Amount of asset is more than max possible";
+}
+
+pub struct WellKnownPriceConverter;
+
+impl WellKnownPriceConverter {
+	pub fn get_ratio(asset_id: CurrencyId) -> Option<Rational64> {
+		match asset_id {
+			CurrencyId::KSM => Some(rational!(375 / 1_000_000)),
+			CurrencyId::ibcDOT => Some(rational!(2143 / 1_000_000)),
+			CurrencyId::USDT | CurrencyId::USDC => Some(rational!(15 / 1_000_000_000)),
+			CurrencyId::aUSD | CurrencyId::kUSD => Some(rational!(15 / 1_000)),
+			CurrencyId::PICA => Some(rational!(1 / 1)),
```

Comment:
I don't see why this stuff is in the "common", it should be runtime specific.

Reply:
Or configurable at least

Reply:
Best place is probably the assets-registry it's self

Reply:
it is common for dali and for picasso. next improvements will be done by using consts overrides and/or trait with impl overrides. also there is idea to have other runtime without relay - so imho it is good enough. 

this code hard to move to pallet as of now.

Reply:
I agree that these conversions will eventually need to be specific to the runtime/native-token, but we are already giving Asset Registry priority in providing our converted ED when we first call `	AssetsRegistry::existential_deposit(*currency_id)` in `fn multi_existential_deposits`

Reply:
sure can do it later. right now dali and picasso use it as common.